cloudformation/database/serverless-database.yaml

```diff
@@ -0,0 +1,345 @@
+    Type: 'AWS::RDS::DBCluster'
+    Properties:
+      Engine: aurora-mysql
+      CopyTagsToSnapshot: true
+      DBClusterParameterGroupName: !Ref ServerlessDatabaseClusterParameterGroup
+      DBSubnetGroupName: !Ref ServerlessDatabaseClusterSubnetGroup
+      EnableHttpEndpoint: false
+      EngineMode: serverless
+      MasterUsername: !Ref ClusterAdmin
+      MasterUserPassword: !Sub '{{resolve:secretsmanager:${AdminPasswordSecret}:SecretString:::}}'
+      ScalingConfiguration:
+        AutoPause: true
+        MaxCapacity: !Ref MaxCapacity
+        MinCapacity: !Ref MinCapacity
+        SecondsUntilAutoPause: 600
+      StorageEncrypted: true
+      Tags:
+        - Key: 'parallel-cluster:accounting'
+          Value: database
+        - Key: 'parallel-cluster:accounting:scheduler'
+          Value: slurm
+        - Key: 'parallel-cluster:accounting:version'
+          Value: '1.0'
+      VpcSecurityGroupIds:
+        - !GetAtt
+          - ServerlessDatabaseClusterSecurityGroup
+          - GroupId
+    UpdateReplacePolicy: Snapshot
+    DeletionPolicy: Snapshot
+  ServerlessClusterDatabaseClientSecurityGroup:
+    Type: 'AWS::EC2::SecurityGroup'
+    Properties:
+      GroupDescription: Security Group to allow connection to Serverless DB Cluster
+      Tags:
+        - Key: 'parallel-cluster:accounting'
+          Value: client-security-group
+        - Key: 'parallel-cluster:accounting:scheduler'
+          Value: slurm
+        - Key: 'parallel-cluster:accounting:version'
+          Value: '1.0'
+      VpcId: !If [CreateVPC, !Ref ServerlessDatabaseClusterVpc, !Ref Vpc]
+  ServerlessClusterDatabaseClientSecurityGroupOutboundRule:
+    Type: 'AWS::EC2::SecurityGroupEgress'
+    Properties:
+      GroupId: !GetAtt
+        - ServerlessClusterDatabaseClientSecurityGroup
+        - GroupId
+      IpProtocol: tcp
+      Description: Allow incoming connections from PCluster
+      DestinationSecurityGroupId: !GetAtt
+        - ServerlessDatabaseClusterSecurityGroup
+        - GroupId
+      FromPort: !GetAtt
+        - ServerlessDatabaseCluster
+        - Endpoint.Port
+      ToPort: !GetAtt
+        - ServerlessDatabaseCluster
+        - Endpoint.Port
+Outputs:
+  parallelclusterdatabasehost:
+    Value: !GetAtt
+      - ServerlessDatabaseCluster
+      - Endpoint.Address
+    Export:
+      Name: !Sub 'parallel-cluster:accounting:database:hostname:${AWS::StackName}'
+  parallelclusterdatabaseport:
+    Value: !GetAtt
+      - ServerlessDatabaseCluster
+      - Endpoint.Port
+    Export:
+      Name: !Sub 'parallel-cluster:accounting:database:port:${AWS::StackName}'
+  parallelclusterdatabaseadminuser:
+    Value: !Ref ClusterAdmin
+    Export:
+      Name: !Sub 'parallel-cluster:accounting:database:admin-user:${AWS::StackName}'
+  parallelclusterdatabasesecretsname:
+    Value: !Ref AdminPasswordSecret
+    Export:
+      Name: !Sub 'parallel-cluster:accounting:database:secrets:name:${AWS::StackName}'
+  parallelclusterdatabasevpc:
```

Comment:
I think it could be useful to also have the Subnets as stack outputs.

Reply:
I was wondering about this. My thinking was that these subnets should be specific and private to the database and not something really exposed for use. For the outputs, I was trying to expose everything that was needed to configure Slurm accounting in the pcluster config file. What is the reason for exposing the subnets?

Reply:
I have not been deploying them into the same subnets. I think it is only necessary (at least without doing some amount more work) that the cluster and database reside in the same VPC. Of course, I may be missing something here.